packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -805,7 +805,9 @@ class _RenderDecoration extends RenderBox {
       + contentPadding.bottom;
     containerHeight = math.max(
       containerHeight,
-      math.max(_boxSize(suffixIcon).height, _boxSize(prefixIcon).height));
+      math.max(
+        _boxSize(suffixIcon).height - (contentPadding.top + contentPadding.bottom),
```

Comment:
You could use `contentPadding.along(Axis.vertical)` instead of the sum (and etc) although it's no biggie.